### ISSUE TITLE
"Run Pester Tests" results in Running ... 'Powershell.RunPesterTestsFromFile' failed

### ISSUE THREAD

**Issue by johndog** [PowerShell/vscode-powershell#2715]
Issue Description
=====

I am experiencing a problem with running pester tests from the Action menu (Control+Shift+P).  When choosing "Run Pester Tests" from this menu on a .tests.ps1 file, pester tests fail to run.  An error is presented in a toast, "'Powershell.RunPesterTestsFromFile' failed".

Attached Logs
=====

The powershell extension log has nothing obviously useful. However, exthost.log has this:

```
[2020-05-20 16:36:18.019] [exthost] [error] TypeError: Cannot read property 'toString' of undefined
    at PesterTestsFeature.launchAllTestsInActiveEditor (c:\Users\johndog\.vscode\extensions\ms-vscode.powershell-preview-2020.5.0\out\src\features\PesterTests.js:47:35)
    at c:\Users\johndog\.vscode\extensions\ms-vscode.powershell-preview-2020.5.0\out\src\features\PesterTests.js:29:18
    at e._executeContributedCommand (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:627:894)
    at e.$executeContributedCommand (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:628:295)
    at e._doInvokeHandler (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:788:757)
    at e._invokeHandler (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:788:449)
    at e._receiveRequest (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:787:120)
    at e._receiveOneMessage (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:785:970)
    at c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:784:213
    at e.fire (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:46:67)
    at v.fire (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:232:615)
    at c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:957:179
    at e.fire (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:46:67)
    at v.fire (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:232:615)
    at t.PersistentProtocol._receiveMessage (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:237:17)
    at c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:234:155
    at e.fire (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:46:67)
    at p.acceptChunk (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:230:129)
    at c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:229:483
    at Socket.t (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:238:565)
    at Socket.emit (events.js:203:13)
    at addChunk (_stream_readable.js:295:12)
    at readableAddChunk (_stream_readable.js:276:11)
    at Socket.Readable.push (_stream_readable.js:210:10)
    at Pipe.onStreamRead (internal/stream_base_commons.js:166:17) PowerShell.RunPesterTestsFromFile
```

Environment Information
=====

Visual Studio Code
-----

| Name | Version |
| --- | --- |
| Operating System | Windows_NT x64 10.0.19628 |
| VSCode | 1.45.1|
| PowerShell Extension Version | 2020.5.0 |

PowerShell Information
-----

|Name|Value|
|---|---|
|PSVersion|7.1.0-preview.3|
|PSEdition|Core|
|GitCommitId|7.1.0-preview.3|
|OS|Microsoft Windows 10.0.19628|
|Platform|Win32NT|
|PSCompatibleVersions|1.0 2.0 3.0 4.0 5.0 5.1.10032.0 6.0.0 6.1.0 6.2.0 7.0.0 7.1.0-preview.3|
|PSRemotingProtocolVersion|2.3|
|SerializationVersion|1.1.0.1|
|WSManStackVersion|3.0|

Visual Studio Code Extensions
-----

<details><summary>Visual Studio Code Extensions(Click to Expand)</summary>

|Extension|Author|Version|
|---|---|---|
|code-runner|formulahendry|0.10.0|
|code-spell-checker|streetsidesoftware|1.9.0|
|cpptools|ms-vscode|0.28.0|
|devinsights|Microsoft|2020.5.7-2|
|gitlens|eamodio|10.2.1|
|indent-one-space|usernamehw|0.2.6|
|line-erase-game|taiki|0.5.2|
|markdown-all-in-one|yzhang|2.8.0|
|perl-outline|hitode909|0.0.6|
|powershell-preview|ms-vscode|2020.5.0|
|rainbow-csv|mechatroner|1.7.0|
|rewrap|stkb|1.10.1|
|show-offset|ramyaraoa|0.0.4|
|unique-lines|bibhasdn|1.0.0|
|vscode-edit-csv|janisdd|0.2.7|
|vscode-markdown-paste-image|telesoho|0.12.3|
|vscode-markdownlint|DavidAnson|0.35.2|
|vscode-nmake-tools|Microsoft|3.0.200420008|

**Comment by SydneyhSmith** [comment]
@johndog thanks for opening this issue, does this reproduce when you instead run the test by hitting F5?

**Comment by johndog** [comment]
Indeed, F5 and the "Run Tests" link on a Describe block both work.

Why does F5 work?  I thought Invoke-Pester was required.

**Comment by TylerLeonhardt** [comment]
My guess is that the Describe block kicks Pester in.

**Comment by SydneyhSmith** [comment]
Thanks @TylerLeonhardt and @johndog looks like we just need to make an addition here: 
https://github.com/PowerShell/vscode-powershell/blob/master/package.json#L298-L302

**Comment by SydneyhSmith** [comment]
Since F5 and the link on the describe block both work, we can hide this in the cmd pallet by adding:

```
 "command": "PowerShell.RunPesterTestsFromFile",
"when": "false"
```

**Comment by johndog** [comment]
There are two problems with F5:
1. F5 always attaches the debugger.  
1. It's not the same...  running a script from the "Run Pester Tests" context menu on a file shows this output:

```
Pester v4.10.1
Executing all tests in 'c:\repos\devdir\build.tests\IncBuild.tests.ps1'

Executing script c:\repos\devdir\build.tests\IncBuild.tests.ps1
```

whereas F5 output just jumps right into the middle, with `Describe` output.

The first issue is the bigger one; if I've set up breakpoints, I don't want to have to turn them all off to run without the debugger.

The second issue is somewhat aesthetic for now, but if Invoke-Pester eventually establishes context that's required before the first Describe command, F5 will stop working entirely.

So it seems there is no formal way to run a command that runs all the pester tests in the current file that's accessible from the keyboard.

**Comment by TylerLeonhardt** [comment]
I've fixed this in #2730. Now the `Run Pester Tests` runs the Pester tests in the file.

**Comment by TylerLeonhardt** [comment]
It's available in the 2020.6.0 version of PowerShell Preview extension.
